validation/src/siwx_message.rs

```diff
@@ -0,0 +1,258 @@
+use std::str::FromStr;
+
+use anyhow::{Context, Result};
+use ssi::{caip10::BlockchainAccountId, caip2::ChainId};
+
+// use ssi_caips::{caip10::BlockchainAccountId, caip2::ChainId};
+use crate::cacao::Capability;
+
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub struct SiwxMessage<'a> {
+    ///RFC 4501 dns authority that is requesting the signing.
+    domain: &'a str,
+    /// Address performing the signing conformant to capitalization
+    /// encoded checksum specified in EIP-55 where applicable if Ethereum.
+    address: String,
```

Comment:
Not a huge fan of the inconsistence between `&'a str` and `String`. Seems like the owned fields are coming from the account, so maybe split this up?

Reply:
Agreed.. I'll see what I can do about nesting owned things. I took this straight from js-did in terms of format. I tried not to clone the entire cacao and borrow what I could easily, but I can't extend the lifetimes of strings I have to generate from it and didn't feel like adding `'b` (or using a bunch of `Option<'&a str>`... it's internal details at this point and only used to generate the siwx message, which ends up allocating a bunch of strings anyway.

Reply:
used all borrows except for account (replaced address/chain_id) and kept DateTime references that are converted for the message